src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
+			IBlockAccess facadeAccess = new FacadeBlockAccess( parentWorld, pos, side, blockState );
+
+			BlockRendererDispatcher dispatcher = Minecraft.getMinecraft().getBlockRendererDispatcher();
+
 			try
 			{
-				return new TextureAtlasAndTint( blockModel.getParticleTexture(), -1 );
+				blockState = blockState.getActualState( facadeAccess, pos );
 			}
-			catch( Exception e )
+			catch( Exception ignored )
 			{
-				if( warnedFor.add( state.getBlock().getRegistryName() ) )
-				{
-					AELog.warn( "Unable to get facade sprite particle texture fallback for blockstate %s. Supressing further warnings for this block.", state );
-					AELog.debug( e );
-				}
 			}
-		}
-
-		return firstFound;
-	}
-
-	private void addFacade( BlockRenderLayer layer, Map<EnumFacing, FacadeRenderState> facades, EnumFacing side, AEAxisAlignedBB busBounds, boolean thinFacades, boolean renderStilt, long rand, CubeBuilder builder )
-	{
-
-		FacadeRenderState facadeState = facades.get( side );
-		IBlockState blockState = facadeState.getSourceBlock();
-
-		builder.setDrawFaces( EnumSet.allOf( EnumFacing.class ) );
-
-		// Reset to no color multiplicator
-		builder.setColorRGB( 0xFFFFFF );
-		builder.useStandardUV();
-
-		// We only render the stilt if we don't intersect with any part directly, and if there's no part on our side
-		if( renderStilt && busBounds == null && layer == BlockRenderLayer.CUTOUT )
-		{
-			builder.setTexture( this.facadeTexture );
-			switch( side )
+			IBakedModel model = dispatcher.getModelForState( blockState );
+			try
 			{
-				case DOWN:
-					builder.addCube( 7, 1, 7, 9, 6, 9 );
-					break;
-				case UP:
-					builder.addCube( 7, 10, 7, 9, 15, 9 );
-					break;
-				case NORTH:
-					builder.addCube( 7, 7, 1, 9, 9, 6 );
-					break;
-				case SOUTH:
-					builder.addCube( 7, 7, 10, 9, 9, 15 );
-					break;
-				case WEST:
-					builder.addCube( 1, 7, 7, 6, 9, 9 );
-					break;
-				case EAST:
-					builder.addCube( 10, 7, 7, 15, 9, 9 );
-					break;
+				blockState = blockState.getBlock().getExtendedState( blockState, facadeAccess, pos );
 			}
-		}
-
-		// Do not add the translucent facade in any other layer than translucent
-		boolean translucent = AEApi.instance().partHelper().getCableRenderMode().transparentFacades;
-		if( translucent && layer != BlockRenderLayer.TRANSLUCENT )
-		{
-			return;
-		}
-
-		final float thickness = thinFacades ? 1 : 2;
-
-		BlockRendererDispatcher blockRendererDispatcher = Minecraft.getMinecraft().getBlockRendererDispatcher();
-		IBakedModel blockModel = blockRendererDispatcher.getModelForState( blockState );
-
-		final int color;
-		if( translucent )
-		{
-			color = 0x4CFFFFFF;
-		}
-		else
-		{
-			color = 0xFFFFFFFF;
-		}
-
-		// TODO: Cache this
-		for( EnumFacing facing : facadeState.getOpenFaces() )
-		{
-			TextureAtlasAndTint spriteAndTint = getSprite( blockModel, blockState, facing, rand );
-			if( spriteAndTint != null && spriteAndTint.sprite != null )
+			catch( Exception ignored )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)